doc/book/reference/usage-examples.md

```diff
@@ -276,81 +276,86 @@ In `config/config.php`, place the following:
+    $config = ArrayUtils::merge($config, include $file);
 }
-return ConfigFactory::fromFiles($files);
+
+// Return an ArrayObject so we can inject the config as a service in Aura.Di
+// and still use array checks like ``is_array``.
+return new ArrayObject($config, ArrayObject::ARRAY_AS_PROPS);
 ```
 
-In `config/autoload/global.php`, place the following:
+In `config/container.php`, place the following:
 
 ```php
 <?php
-return [
-    'routes' => [
-        [
-            'path' => '/',
-            'middleware' => 'Application\HelloWorld',
-            'allowed_methods' => [ 'GET' ],
-        ],
-        [
-            'path' => '/ping',
-            'middleware' => 'Application\Ping',
-            'allowed_methods' => [ 'GET' ],
-        ],
-    ],
-];
-```
 
-In `config/dependencies.php`, place the following:
+use Zend\ServiceManager\Config;
+use Zend\ServiceManager\ServiceManager;
 
-```php
-<?php
-use Zend\Config\Factory as ConfigFactory;
-use Zend\Stdlib\Glob;
+// Load configuration
+$config = require __DIR__.'/config.php';
 
-return ConfigFactory::fromFiles(
-    Glob::glob('config/autoload/dependencies.{global,local}.php', Glob::GLOB_BRACE)
-);
+// Build container
+$container = new ServiceManager();
+(new Config($config['dependencies']))->configureServiceManager($container);
+
+// Inject config
+$container->setService('config', $config);
+
+return $container;
 ```
 
 In `config/autoload/dependencies.global.php`, place the following:
 
 ```php
 <?php
+
 return [
-    'services' => [
-        'config' => include __DIR__ . '/../config.php',
-    ],
-    'invokables' => [
-        'Application\HelloWorld' => 'Application\HelloWorld',
-        'Application\Ping' => 'Application\Ping',
-    ],
-    'factories' => [
-        'Zend\Expressive\Application' => 'Zend\Expressive\Container\ApplicationFactory',
-    ],
+    'dependencies' => [
+        'invokables' => [
+            \Application\HelloWorldAction::class => \Application\HelloWorldAction::class,
+            \Application\PingAction::class => \Application\PingAction::class,
+        ],
+        'factories' => [
+            \Zend\Expressive\Application::class => \Zend\Expressive\Container\ApplicationFactory::class,
+        ],
+    ]
 ];
 ```
 
-In `config/services.php`, place the following:
+In `config/autoload/routes.global.php`, place the following:
 
 ```php
 <?php
-use Zend\ServiceManager\Config;
-use Zend\ServiceManager\ServiceManager;
 
-return new ServiceManager(new Config(include 'config/dependencies.php'));
+return [
+    'routes' => [
+        [
+            'path' => '/',
+            'middleware' => \Application\HelloWorldAction::class,
+            'allowed_methods' => [ 'GET' ],
+        ],
+        [
+            'path' => '/ping',
+            'middleware' => \Application\PingAction::class,
+            'allowed_methods' => [ 'GET' ],
+        ],
+    ],
+];
 ```
 
-In `src/HelloWorld.php`, place the following:
+In `src/Application/HelloWorld.php`, place the following:
```

Comment:
That's not the correct spot to make that change. I had a further look and needed to change App to Application in the class files (Ping, HelloWorld) to match the rest of the code.